setup.py

```diff
@@ -27,13 +27,13 @@
 
 install_requires = [
     "craft-cli",
-    "craft-parts",
+    "craft-parts>=1.18",
```

Comment:
interesting; how come these are necessary?

Reply:
For craft-parts: Not entirely sure, but one of my environments had craft-parts 1.17 and stuff broke

For craft-store: 2.4 introduces the `unregister_name` method on the store client

For pydantic: Something got me 1.8.2 for some reason in an environment and it broke things. *technically* we need pydantic 1.10+ since craft-store depends on it, but if we fall back to an earlier craft-store (understanding that unregistering won't work) charmcraft still works fine.